### PR TITLE
Added description to ZIS return codes

### PR DESCRIPTION
**Pull request by adityadr-joy**
Signed-off-by: Aditya Ranshinge <aranshinge@rocketsoftware.com>

Added description to ZIS return codes

### CODE REVIEW

File: c/crossmemory.c [hunk 1]
```diff
@@ -71,13 +71,88 @@ const CrossMemoryServerName CMS_DEFAULT_SERVER_NAME = {CROSS_MEMORY_DEFAULT_SERV
...
+  [RC_CMS_MAIN_LOOP_FAILED] = "Main loop unexpectedly terminated",
+  [RC_CMS_IMPROPER_SERVICE_AS] = "Invalid service address space",
+  [RC_CMS_BAD_SERVER_KEY] = "Core server started with wrong key", 
+  [RC_CMS_RESMGR_NOT_ADDED] = "RESMGR not added",
+  [RC_CMS_RESMGR_NOT_REMOVED] = "RESMGR not removed",
+  [RC_CMS_RESMGR_NOT_LOCKED] = "RESMGR not locked",
+  [RC_CMS_RESMGR_NAMETOKEN_FAILED] = "RESMGR Name/Token not retrieved",
   [RC_CMS_WRONG_SERVER_VERSION] = "Wrong server version",
+  [RC_CMS_LATENT_PARM_NULL] = "Latent parameter is NULL",
+  [RC_CMS_USER_PARM_NULL] = "User parameter is NULL",
```
Comment (ifakhrutdinov) [inline c/crossmemory.c:122]: `Cross-memory server parameter list is NULL`

File: c/crossmemory.c [hunk 2]
```diff
@@ -71,13 +71,88 @@ const CrossMemoryServerName CMS_DEFAULT_SERVER_NAME = {CROSS_MEMORY_DEFAULT_SERV
...
+  [RC_CMS_PC_HDLR_PARM_BAD_EYECATCHER] = "PC handler parameter has invalid eyecatcher",
```
Comment (ifakhrutdinov) [inline c/crossmemory.c:123]: `PC handler parameter list has an invalid eyecatcher`

File: c/crossmemory.c [hunk 3]
```diff
@@ -71,13 +71,88 @@ const CrossMemoryServerName CMS_DEFAULT_SERVER_NAME = {CROSS_MEMORY_DEFAULT_SERV
...
+  [RC_CMS_ZERO_PC_NUMBER] = "Zero PC number",
```
Comment (ifakhrutdinov) [inline c/crossmemory.c:124]: `PC is unavailable`

I know this does not exactly match the name but that's usually what that means. It's zeroed out when someone cancels ZIS.

File: c/crossmemory.c [hunk 4]
```diff
@@ -71,13 +71,88 @@ const CrossMemoryServerName CMS_DEFAULT_SERVER_NAME = {CROSS_MEMORY_DEFAULT_SERV
...
   [RC_CMS_WRONG_CLIENT_VERSION] = "Wrong client version",
+  [RC_CMS_ZVTE_CHAIN_LOOP] = "",
```
Comment (ifakhrutdinov) [inline c/crossmemory.c:126]: `Potential loop in the ZVTE chain detected`

File: c/crossmemory.c [hunk 5]
```diff
@@ -71,13 +71,88 @@ const CrossMemoryServerName CMS_DEFAULT_SERVER_NAME = {CROSS_MEMORY_DEFAULT_SERV
...
+  [RC_CMS_ZVTE_CHAIN_NOT_LOCKED] = "ZVTE chain is not locked", 
+  [RC_CMS_ZVTE_CHAIN_NOT_RELEASED] = "ZVTE chain is not released",
+  [RC_CMS_SNPRINTF_FAILED] = "snprintf failed",
+  [RC_CMS_SLH_NOT_CREATED] = "Short lived heap is not created",
+  [RC_CMS_MSG_QUEUE_NOT_CREATED] = "Message queue is not created",
+  [RC_CMS_CONFIG_HT_NOT_CREATED] = "Server config hash table is not created",
+  [RC_CMS_SLH_ALLOC_FAILED] = "Short lived heap memory allocation failed",
+  [RC_CMS_UNKNOWN_PARM_TYPE] = "Config parameter type is not known",
```
Comment (ifakhrutdinov) [inline c/crossmemory.c:134]: `Unknown config parameter type`

File: c/crossmemory.c [hunk 6]
```diff
@@ -71,13 +71,88 @@ const CrossMemoryServerName CMS_DEFAULT_SERVER_NAME = {CROSS_MEMORY_DEFAULT_SERV
...
+  [RC_CMS_CONFIG_PARM_NAME_TOO_LONG] = "Config parameter key name is too long",
+  [RC_CMS_CHAR_PARM_TOO_LONG] = "Value of config parameter is too long",
+  [RC_CMS_CONFIG_PARM_NOT_FOUND] = "Config parameter not found",
+  [RC_CMS_STDSVC_PARM_NULL] = "Service parameter is NULL",
+  [RC_CMS_STDSVC_PARM_BAD_EYECATCHER] = "Service parameter has invalid eyecatcher",
```
Comment (ifakhrutdinov) [inline c/crossmemory.c:139]: `Service parameter list has an invalid eyecatcher`

File: c/crossmemory.c [hunk 7]
```diff
@@ -71,13 +71,88 @@ const CrossMemoryServerName CMS_DEFAULT_SERVER_NAME = {CROSS_MEMORY_DEFAULT_SERV
...
+  [RC_CMS_SNPRINTF_FAILED] = "snprintf failed",
+  [RC_CMS_SLH_NOT_CREATED] = "Short lived heap is not created",
+  [RC_CMS_MSG_QUEUE_NOT_CREATED] = "Message queue is not created",
+  [RC_CMS_CONFIG_HT_NOT_CREATED] = "Server config hash table is not created",
+  [RC_CMS_SLH_ALLOC_FAILED] = "Short lived heap memory allocation failed",
+  [RC_CMS_UNKNOWN_PARM_TYPE] = "Config parameter type is not known",
+  [RC_CMS_CONFIG_PARM_NAME_TOO_LONG] = "Config parameter key name is too long",
+  [RC_CMS_CHAR_PARM_TOO_LONG] = "Value of config parameter is too long",
+  [RC_CMS_CONFIG_PARM_NOT_FOUND] = "Config parameter not found",
+  [RC_CMS_STDSVC_PARM_NULL] = "Service parameter is NULL",
```
Comment (ifakhrutdinov) [inline c/crossmemory.c:138]: `Service parameter list is NULL`

File: c/crossmemory.c [hunk 8]
```diff
@@ -71,13 +71,88 @@ const CrossMemoryServerName CMS_DEFAULT_SERVER_NAME = {CROSS_MEMORY_DEFAULT_SERV
...
+  [RC_CMS_STDSVC_PARM_BAD_EYECATCHER] = "Service parameter has invalid eyecatcher",
+  [RC_CMS_NOT_APF_AUTHORIZED] = "Not APF-authorized",
```
Comment (ifakhrutdinov) [inline c/crossmemory.c:140]: `Cross-memory server is not APF-authorized`

File: c/crossmemory.c [hunk 9]
```diff
@@ -71,13 +71,88 @@ const CrossMemoryServerName CMS_DEFAULT_SERVER_NAME = {CROSS_MEMORY_DEFAULT_SERV
...
+  [RC_CMS_NO_STEPLIB] = "STEPLIB not found",  
+  [RC_CMS_MODULE_NOT_IN_STEPLIB] = "Module not found in STEPLIB",
```
Comment (ifakhrutdinov) [inline c/crossmemory.c:142]: `Cross-memory server module not found in STEPLIB`

File: c/crossmemory.c [hunk 10]
```diff
@@ -71,13 +71,88 @@ const CrossMemoryServerName CMS_DEFAULT_SERVER_NAME = {CROSS_MEMORY_DEFAULT_SERV
...
   [RC_CMS_SERVER_NAME_NULL] = "Server name is NULL",
+  [RC_CMS_SERVICE_ENTRY_OCCUPIED] = "Service entry occupied",
+  [RC_CMS_NO_STORAGE_FOR_MSG] = "No storage available for msg",
```
Comment (ifakhrutdinov) [inline c/crossmemory.c:145]: `No storage in the message queue`

File: c/crossmemory.c [hunk 11]
```diff
@@ -71,13 +71,88 @@ const CrossMemoryServerName CMS_DEFAULT_SERVER_NAME = {CROSS_MEMORY_DEFAULT_SERV
...
+  [RC_CMS_CONFIG_PARM_NAME_TOO_LONG] = "Config parameter key name is too long",
+  [RC_CMS_CHAR_PARM_TOO_LONG] = "Value of config parameter is too long",
+  [RC_CMS_CONFIG_PARM_NOT_FOUND] = "Config parameter not found",
+  [RC_CMS_STDSVC_PARM_NULL] = "Service parameter is NULL",
+  [RC_CMS_STDSVC_PARM_BAD_EYECATCHER] = "Service parameter has invalid eyecatcher",
+  [RC_CMS_NOT_APF_AUTHORIZED] = "Not APF-authorized",
+  [RC_CMS_NO_STEPLIB] = "STEPLIB not found",  
+  [RC_CMS_MODULE_NOT_IN_STEPLIB] = "Module not found in STEPLIB",
   [RC_CMS_SERVER_NAME_NULL] = "Server name is NULL",
+  [RC_CMS_SERVICE_ENTRY_OCCUPIED] = "Service entry occupied",
```
Comment (ifakhrutdinov) [inline c/crossmemory.c:144]: `Service entry already occupied`

**Review by ifakhrutdinov — COMMENTED** [review]
I think it's a good first version. We can refine the descriptions in the future. Thanks!

**Comment by adityadr-joy** [comment]
Resolved all comments given by @ifakhrutdinov . Thanks Irek for the inputs.

**Review by ifakhrutdinov — APPROVED** [review]

**Pull request merged**